src/Serialization/containers.jl

```diff
@@ -28,3 +28,20 @@ function load_internal(s::DeserializerState, T::Type{<:Tuple}, dict::Dict)
 end
 
 
+
+################################################################################
+# Saving and loading matrices
+function save_internal(s::SerializerState, mat::Matrix{T}) where T
+    m, n = size(mat)
+    return Dict(
+        :matrix => [save_type_dispatch(s, [mat[i, j] for j in 1:n]) for i in 1:m]
+    )
+end
+
+function load_internal(s::DeserializerState, ::Type{Matrix{T}}, dict::Dict) where T
+    x = dict[:matrix]
+    y = vcat([permutedims(load_type_dispatch(s, Vector{T}, x[i])) for i in 1:length(x)]...)
```

Comment:
```suggestion
    y = reduce(vcat, [permutedims(load_type_dispatch(s, Vector{T}, x[i])) for i in 1:length(x)])
```
The splatting is expensive in this case.